src/main/java/org/opentripplanner/datastore/file/ZipFileDataSource.java

```diff
@@ -93,6 +96,27 @@ private void loadContent() {
         ZipEntry entry = entries.nextElement();
         content.add(new ZipFileEntryDataSource(this, entry));
       }
+    } catch (ZipException ze) {
+      // Java needs help for standard ZIP files with names encoded in cp437
+      // this allows decoding of utf-8 and cp437 at the same time!
+      // As Cp437 is not guaranteed to be available try it but don't depend on it.
+      // see APPENDIX D - Language Encoding (EFS) in the format documentation
+      // https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
+      // additional Recommendations for Interoperability can be found at Apache Commons
+      // https://commons.apache.org/proper/commons-compress/zip.html
+      try {
+        Charset charset = Charset.forName("Cp437");
+        this.zipFile = new ZipFile(file, ZipFile.OPEN_READ, charset);
+
+        Enumeration<? extends ZipEntry> entries = zipFile.entries();
+
+        while (entries.hasMoreElements()) {
+          ZipEntry entry = entries.nextElement();
+          content.add(new ZipFileEntryDataSource(this, entry));
+        }
```

Comment:
This has been discussed, but I think this should be extracted into a `private static` method, and reused. Something like this:
```
/** 
  * Map ZipEntries into a DataSource. This does not read the content of each zip entry, only create a 
  * {@link DataSource} around it.
  */
private static Collection<DataSource> listZipEntries(ZipFile zipFile) throws IOException/ZipException {
    List<DataSource> content = new ArrayList<DataSource>();
    Enumeration<? extends ZipEntry> entries = zipFile.entries();
    
    while (entries.hasMoreElements()) {
        ZipEntry entry = entries.nextElement();
        content.add(new ZipFileEntryDataSource(this, entry));
    }
    return content;
}
```

Reply:
Note! This also fixes a problem with the current implementation, because one or more entries can be added before the error happens, if the list is cashed this can not happen.